component-samples/demo/scripts/fdorvconfig.sh

```diff
@@ -0,0 +1,172 @@
+        a) aio_path=${OPTARG};;
+        h) Help
+           exit 0;;
+        o) onr_path=${OPTARG};;
+        p) custom_proxy=${OPTARG};;
+        \?) echo "Error: Invalid Option, use -h for help"
+            exit 1;;
+    esac
+done
+
+# Setting log level with color codes
+colgrn='\033[0;32m'
+colred='\033[0;31m'
+colylw='\033[0;33m'
+nocol='\033[0m'
+
+info_level="${colgrn}INFO:${nocol}"
+err_level="${colred}ERROR:${nocol}"
+warn_level="${colylw}WARN:${nocol}"
+
+if [ -z "$aio_path" ] && [ -z "$onr_path" ]; then
+    echo -e "$err_level Path to either AIO or Owner directory is mandatory, check usage with -h" >&2
+    exit 1
+fi
+
+conn_timeout="30"
+extracted_crt="/tmp/extracted_crt"
+target_rv_addr="fdorv.com"
+target_rv_port="443"
+hosted_rvcert="/tmp/hostedrv_crt"
+
+# Downloads hosted RV certificate and adds to ca-cert.pem of AIO and Owner Components.
+# Updating respective worker in AIO to trigger TO0 with hosted RV instance.
+function config_fdo_components() {
+    target_component_dir=$1
+    if [ ! -d "$target_component_dir" ]; then
+        echo -e "$err_level No such directory $target_component_dir exists, please check and try again." >&2
+        exit 1
+    fi
+    if [[ "${target_component_dir: -1}" == "/" ]]; then
+        target_component_dir=${target_component_dir%/}
+    fi
+    secrets_dir="$target_component_dir/secrets"
+    cacert_file="$secrets_dir/ca-cert.pem"
+    if [ ! -d "$secrets_dir" ]; then
+        echo -e "$err_level $secrets_dir directory doesn't exist, please check and try again." >&2
+        exit 1
+    fi
+    if [ ! -f "$cacert_file" ]; then
+        echo -e "$err_level Required file $cacert_file doesn't exist, please check and try again." >&2
+        exit 1
+    fi
+    # Checking if already Hosted RV certificate is present in target ca-cert.pem
+    if [ -f "$hosted_rvcert" ]; then
+        if [ -s "$hosted_rvcert" ]; then
+            cert_lines=`wc -l $hosted_rvcert | cut -d ' ' -f 1`
+            grep -A $cert_lines "END CERTIFICATE" $cacert_file | sed -n '2,$p' > $extracted_crt
+            diff -q $extracted_crt $hosted_rvcert &>/dev/null
+            if [[ $? -eq 0 ]]; then
+                echo -e "$info_level Hosted RV $target_rv_addr certificate is already present in $cacert_file, hence skipping downloading and adding."
+                return
+            fi
+        fi
+    fi
+    # Testing connection to target site and downloading certificate
+    echo -e "$info_level Downloading hosted RV instance, $target_rv_addr certificate to $cacert_file"
+    if [ $required_proxy ]; then
+        proxy_host=$(echo $required_proxy | cut -d '/' -f 3 | cut -d ':' -f 1)
+        proxy_port=$(echo $required_proxy | cut -d '/' -f 3 | cut -d ':' -f 2)
+        proxy_arg="-proxy ${proxy_host}:${proxy_port}"
+    else
+        proxy_arg=""
+    fi
+    timeout --preserve-status $conn_timeout openssl s_client $proxy_arg -connect $target_rv_addr:$target_rv_port 1>/dev/null 2>/dev/null </dev/null
```

Comment:
1>/dev/null  - stdout to dev/null
2>/dev/null  - stderr to dev/null
</dev/null     - What's the purpose ?

Reply:
This command waits for input after execution, hence `< /dev/null` is used to provide `EOF`